lms/static/js/learner_dashboard/views/course_entitlement_view.js

```diff
@@ -0,0 +1,309 @@
+                         html: true,
+                         trigger: 'click',
+                         content: '<div class="verification-modal" role="dialog"' +
+                             'aria-labelledby="enrollment-verification-title">' +
+                             '<p id="enrollment-verification-title">' + confirmationHTML + '</p>' +
+                             '<div class="action-items">' +
+                             '<button type="button" class="popover-dismiss final-confirmation-btn">' +
+                             gettext('No') + '</button>' +
+                             '<button type="button" class="enroll-btn final-confirmation-btn">' +
+                             gettext('Yes') + '</button></div></div>'
+                     });
+
+                     // Close popover on click-away
+                     $(document).on('click', function(e) {
+                         if (!($(e.target).closest('.enroll-btn-initial, .popover').length)) {
+                             this.$('.enroll-btn-initial').popover('hide');
+                         }
+                     }.bind(this));
+
+                     // Ensure that focus moves to the popover on click of the initial change enrollment button.
+                     this.$('.enroll-btn-initial').on('click', function() {
+                         this.$('.final-confirmation-btn:first').focus();
+                     }.bind(this));
+                 },
+
+                 handleVerificationPopoverA11y: function(e) {
+                     /* Ensure that the second step verification popover is treated as an a11y compliant dialog */
+                     var nextButton,
+                         openButton = $(e.target).closest('.course-entitlement-selection-container')
+                             .find('.enroll-btn-initial');
+                     if (e.key === 'Tab') {
+                         e.preventDefault();
+                         nextButton = $(e.target).is(':first-child') ? $(e.target).next('.final-confirmation-btn') :
+                             $(e.target).prev('.final-confirmation-btn');
+                         nextButton.focus();
+                     } else if (e.key === 'Escape') {
+                         openButton.popover('hide');
+                         openButton.focus();
+                     }
+                 },
+
+                 hideVerificationDialog: function() {
+                     this.$('.enroll-btn-initial').focus().popover('hide');
+                 },
+
+                 formatCourseHomeUrl: function(sessionKey) {
+                     /*
+                     Takes the base course home URL and updates it with the new session id, leveraging the
+                     the fact that all course keys contain a '+' symbol.
+                     */
+                     var oldSessionKey = this.courseHomeUrl.split('/').filter(function(x) { return x.indexOf('+') > 0; })[0];  // eslint-disable-line max-len
```

Comment:
Style nitpick: this is difficult to read, it would be clearer across multiple lines.

Reply:
Changed